astarte-device-sdk/AstarteDeviceSDK.cpp

```diff
@@ -373,3 +373,13 @@ AstarteDeviceSDK::ConnectionStatus AstarteDeviceSDK::connectionStatus() const
 
     return static_cast<AstarteDeviceSDK::ConnectionStatus>(m_astarteTransport->connectionStatus());
 }
+
+bool AstarteDeviceSDK::connectToBroker()
+{
+  return m_astarteTransport->connectToBroker();
```

Comment:
`m_astarteTransport` is initialized asynchronously, so you should always check if it's `null` before calling methods on it